std/algorithm/iteration.d

```diff
@@ -902,67 +913,100 @@ template each(alias pred = "a")
             size_t i = 0;
             while (!r.empty)
             {
-                cast(void) binaryFun!BinaryArgs(i, r.front);
+                static if (!is(typeof(binaryFun!BinaryArgs(i, r.front)) ==
+                        Flag!"each"))
+                    cast(void) binaryFun!BinaryArgs(i, r.front);
```

Comment:
It's required in D if `binaryFun` is a strongly pure function and has a return type of `void`.